pallets/ah-migrator/src/vesting.rs

```diff
@@ -86,3 +89,64 @@ pub mod alias {
 		V1,
 	}
 }
+
+#[cfg(feature = "std")]
+impl<T: Config> crate::types::AhMigrationCheck for VestingMigrator<T> {
+	type RcPrePayload =
+		Vec<(Vec<u8>, Vec<BalanceOf<T>>, Vec<GenericVestingInfo<BlockNumberFor<T>, BalanceOf<T>>>)>;
+
+	type AhPrePayload =
+		Vec<(Vec<u8>, Vec<BalanceOf<T>>, Vec<GenericVestingInfo<BlockNumberFor<T>, BalanceOf<T>>>)>;
+
+	fn pre_check(_: Self::RcPrePayload) -> Self::AhPrePayload {
+		pallet_vesting::Vesting::<T>::iter()
+			.map(|(who, schedules)| {
+				let balances: Vec<BalanceOf<T>> = schedules.iter().map(|s| s.locked()).collect();
+				let vesting_info: Vec<GenericVestingInfo<BlockNumberFor<T>, BalanceOf<T>>> =
+					schedules
+						.iter()
+						.map(|s| GenericVestingInfo {
+							locked: s.locked(),
+							starting_block: s.starting_block(),
+							per_block: s.per_block(),
+						})
+						.collect();
+				(who.encode(), balances, vesting_info)
+			})
+			.collect()
+	}
+
+	fn post_check(rc_pre_payload: Self::RcPrePayload, ah_pre_payload: Self::AhPrePayload) {
+		use std::collections::BTreeMap;
+
+		let all_pre: BTreeMap<
+			Vec<u8>,
+			(Vec<BalanceOf<T>>, Vec<GenericVestingInfo<BlockNumberFor<T>, BalanceOf<T>>>),
+		> = rc_pre_payload
+			.into_iter()
+			.chain(ah_pre_payload.into_iter())
+			.map(|(who, balances, vesting_info)| (who, (balances, vesting_info)))
+			.collect();
+
+		let all_post: BTreeMap<
+			Vec<u8>,
+			(Vec<BalanceOf<T>>, Vec<GenericVestingInfo<BlockNumberFor<T>, BalanceOf<T>>>),
+		> = pallet_vesting::Vesting::<T>::iter()
+			.map(|(who, schedules)| {
+				let balances: Vec<BalanceOf<T>> = schedules.iter().map(|s| s.locked()).collect();
+				let vesting_info: Vec<GenericVestingInfo<BlockNumberFor<T>, BalanceOf<T>>> =
+					schedules
+						.iter()
```

Comment:
It can be a bit easier to read if you first create a vector and then `push` into that, but it doesnt really matter for test code.